### PR TITLE
Lib refact

### PR DESCRIPTION
**Pull request by amine7536**
Use more common project structure :
- `cmd`: application command line
- `pkg`: split `lib` into `pkg/slack`, `pkg/homoglyph`, `pkg/model` …
- `config`: initialize application configuration

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,15 +9,15 @@ WORKDIR /src
...
-RUN go build -ldflags="-s -w" -o cercat
+RUN go build -ldflags="-s -w" -o cercat ./cmd/cercat
 
 # Final Docker image
 FROM alpine AS final-stage
 LABEL MAINTAINER "Thomas Labarussias <issif+github@gadz.org>"
 
 RUN apk add --no-cache ca-certificates
 
-# Create user falcosidekick
```
Comment (Issif) [inline Dockerfile:20]: We see from which project I took the Dockerfile 🤣

File: lib/lib.go [hunk 2]
```diff
@@ -129,12 +96,12 @@ func isIDN(domain string) bool {
 }
 
 // IsMatchingCert checks if certificate matches the regexp
-func IsMatchingCert(config *Configuration, result *Result, reg *regexp.Regexp) bool {
+func IsMatchingCert(homoglyphs map[string]string, result *model.Result, reg *regexp.Regexp) bool {
```
Comment (Issif) [inline lib/lib.go]: the map is pretty big, can we pass it's pointer for avoiding copy of memory?

File: lib/lib_test.go [hunk 3]
```diff
@@ -44,7 +48,7 @@ var _ = Describe("Handler", func() {
 		msg, _ := ioutil.ReadFile("../res/cert.json")
 		It("should return a valid payload", func() {
 			result, err := lib.ParseResultCertificate(msg)
-			slackPayload := lib.NewSlackPayload(config, result)
+			slackPayload := slack.NewSlackPayload(config, result)
```
Comment (Issif) [inline lib/lib_test.go]: We should rename the method in `slack.NewPayload()`, what do you think?

**Review by Issif — CHANGES_REQUESTED** [review]

File: lib/lib_test.go (hunk 3 above)
Comment (amine7536) [reply lib/lib_test.go]: Yes, I updated `pkg/slack`.

File: lib/lib.go (hunk 2 above)
Comment (amine7536) [reply lib/lib.go]: Yes, it's done. We'll use a pointer to homoglyph map.

**Comment by amine7536** [comment]
I also fixed the test. We should add a Github workflow to run `go test ./...` in another PR.

File: lib/lib.go [hunk 4]
```diff
@@ -14,52 +18,15 @@ import (
...
-	AllDomains   []string               `json:"all_domains"`
-}
-
 // the websocket stream from calidog
 const certInput = "wss://certstream.calidog.io"
 
 // CertCheckWorker parses certificates and raises alert if matches config
-func CertCheckWorker(config *Configuration) {
-	reg, _ := regexp.Compile(config.Regexp)
+func CertCheckWorker(r string, homoglyph map[string]string, msgChan chan []byte, bufferChan chan *model.Result) {
```
Comment (Issif) [inline lib/lib.go]: a pointer to homoglyph map here too

**Review by Issif — CHANGES_REQUESTED** [review]

File: lib/lib.go (hunk 4 above)
Comment (amine7536) [reply lib/lib.go]: Done

**Review by Issif — APPROVED** [review]

**Comment by Issif** [comment]
@amine7536 I merged this PR and added you as collaborator. You can now add the prometheus endpoint and I'll release a new version right when you're ready.

**Pull request merged**